geneate_dynein_conformation_graphs/englishORspanish.py

```diff
@@ -0,0 +1,199 @@
+
+        # Filter out angles that are greater than the angle_threshold
+        if angle_degrees <= angle_threshold: 
+            angles.append(angle_degrees)
+    
+    return angles
+
+# Function to process the angle data 
+def calculate_fit(data_lists, foldername):
+    # Adjustable parameters, after optimizing, these were the best settings
+    angle_threshold = 6  # When the angle is greater than x degrees, it is considered a bend
+    bend_threshold = 100 # Everything greater than x degrees, will be removed because junkdata
+    num_segments = 10    # How many segments per doublet
+
+    # Initialize empty lists
+    all_angles = [] 
+    bent_lists = [] 
+    bending_amounts = []
+    angle_differences = []
+
+    # Loop through each doublet of data points
+    for index, data in enumerate(data_lists):
+        # Calculate angles for the current doublet
+        angles = calculate_angles(data, num_segments, bend_threshold)
+        all_angles.append(angles)
+        
+        # Sum of absolute angles as bending metric
+        bending_amount = np.sum(np.abs(angles))  
+        bending_amounts.append(bending_amount)
+
+        # Filter all angles greater than the angle_threshold
+        if any(angle > angle_threshold for angle in angles):
+            bent_lists.append(index+1)
+
+    # Print results
+    if print_results:
+        for i, (angles, bending_amount) in enumerate(zip(all_angles, bending_amounts)):
+            print(f"Angles for list {i+1} (in degrees): {angles}")
+            print(f"Bending amount for list {i+1}: {bending_amount}")
+
+    # Calculate the aggregate bending scores
+    mean_bending = np.mean(bending_amounts)
+    max_bending = np.max(bending_amounts)
+
+    # Calculate max angle difference
+    for doublet in all_angles:
+        min_angle = np.min(doublet)
+        max_angle = np.max(doublet)
+
+        angle_difference = max_angle - min_angle
+        angle_differences.append(angle_difference)
+
+    angle_differences_average = np.average(angle_differences)
+
+    # Draw a conclusion based on the presence of bends 
+    if bent_lists:
+        result = f"Doublets likely to have a bend: {bent_lists}"
+    else:
+        result = "All doublets are likely straight."
+
+    # Write results to a file in the EORS folder
+    # Note if you want to have the results in Results/poly01Tomo01 folder for example, you should change {results_folder} to {foldername}
+    single_result = f"{foldername}_bending_results.txt"
+    single_result_path = f"{analysis_folder}/{results_folder}/{single_result}"
+
+    with open(single_result_path, "w") as file:
+        for i, (angles, bending_amount, angle_difference) in enumerate(zip(all_angles, bending_amounts, angle_differences)):
+            file.write(f"Angles for doublet {i+1} (in degrees): {angles}\n")
+            file.write(f"Max angle difference: {angle_difference}\n")
+            file.write(f"Bending amount for doublet {i+1}: {bending_amount}\n\n")
```

Comment:
same index comment as before
```suggestion
        for i, (angles, bending_amount, angle_difference) in enumerate(zip(all_angles, bending_amounts, angle_differences), start=1):
            file.write(f"Angles for doublet {i} (in degrees): {angles}\n")
            file.write(f"Max angle difference: {angle_difference}\n")
            file.write(f"Bending amount for doublet {i}: {bending_amount}\n\n")
```